### PR TITLE
[imagetool] Scale down large images before passing to ImageTool

### PR DESCRIPTION
**Pull request by bjoerge**
This will scale down large images before passing on to the ImageTool component. It's a defensive fix to ensure the imagetool works even if you give it a huge image.

This is just an internal refactoring, the public API stays the same.

It is also a good idea to scale down the images before fetching from the CDN, but that's a separate task.

### CODE REVIEW

File: packages/@sanity/imagetool/src/Resize.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+
+export default class ImageLoader extends React.Component {
```
Comment (rexxars) [inline packages/@sanity/imagetool/src/Resize.js]: Incorrect class name

File: packages/@sanity/imagetool/src/Resize.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  static propTypes = {
+    image: PropTypes.instanceOf(HTMLImageElement).isRequired,
+    maxWidth: PropTypes.number.isRequired,
+    maxHeight: PropTypes.number.isRequired,
+    children: PropTypes.func.isRequired
+  }
+
+  getCanvas() {
```
Comment (rexxars) [inline packages/@sanity/imagetool/src/Resize.js]: Won't this leak a new canvas into the document for every `Resize` instance?

Comment (bjoerge) [reply packages/@sanity/imagetool/src/Resize.js]: Whoops, I had implemented a cWU that removed it, but it got missing in action obviously. Added it back now. Thanks!

File: packages/@sanity/imagetool/src/Resize.js (hunk 1 above)
Comment (bjoerge) [reply packages/@sanity/imagetool/src/Resize.js]: Good catch, thanks!

**Pull request merged**